### PR TITLE
Display available venues when search empty

### PR DESCRIPTION
**Pull request by jaco3n4**

### CODE REVIEW

**Comment by mathieugerard** [comment]
The function is correct.
It's a pity that the code is duplicated.
@Sparksx please review.

File: src/header/searchBar/searchBar.ts [hunk 1]
```diff
@@ -104,6 +104,8 @@ export class SearchBar extends DefaultControl {
   private _searchFocus (e: JQueryEventObject): void {
     if (this._map.getVenue()) {
       this._map.headerManager.showSearchResults('mainSearches', this._clickOnSearchResult.bind(this))
+    } else if ((!this._map.getVenue() && !this._map.getDirection())) {
+      this._map.headerManager.search(' ', searchOptions(this._map, this._map.getVenue(), 'search'), this._clickOnSearchResult.bind(this))
```
Comment (Sparksx) [inline src/header/searchBar/searchBar.ts]:
```suggestion
      this._map.headerManager.search(' ', searchOptions(this._map, null, 'search'), this._clickOnSearchResult.bind(this))
```

File: src/header/searchBar/searchBar.ts [hunk 2]
```diff
@@ -104,6 +104,8 @@ export class SearchBar extends DefaultControl {
   private _searchFocus (e: JQueryEventObject): void {
     if (this._map.getVenue()) {
       this._map.headerManager.showSearchResults('mainSearches', this._clickOnSearchResult.bind(this))
+    } else if ((!this._map.getVenue() && !this._map.getDirection())) {
```
Comment (Sparksx) [inline src/header/searchBar/searchBar.ts]:
```suggestion
    } else {
```
we cannot be in direction outside venue, so i think this condition is not really needed. (take time to test with just an else)

File: src/header/searchBar/searchBar.ts [hunk 3]
```diff
@@ -122,6 +124,8 @@ export class SearchBar extends DefaultControl {
         this._map.headerManager.search(searchString, searchOptions(this._map, this._map.getVenue(), 'search'), this._clickOnSearchResult.bind(this))
       } else if (this._map.getVenue()) {
         this._map.headerManager.showSearchResults('mainSearches', this._clickOnSearchResult.bind(this))
+      } else if ((!this._map.getVenue() && !this._map.getDirection())) {
```
Comment (Sparksx) [inline src/header/searchBar/searchBar.ts]:
```suggestion
      } else {
```

File: src/header/searchBar/searchBar.ts [hunk 4]
```diff
@@ -122,6 +124,8 @@ export class SearchBar extends DefaultControl {
...
+        this._map.headerManager.search(' ', searchOptions(this._map, this._map.getVenue(), 'search'), this._clickOnSearchResult.bind(this))
```
Comment (Sparksx) [inline src/header/searchBar/searchBar.ts]:
```suggestion
        this._map.headerManager.search(' ', searchOptions(this._map, null, 'search'), this._clickOnSearchResult.bind(this))
```

**Pull request merged**